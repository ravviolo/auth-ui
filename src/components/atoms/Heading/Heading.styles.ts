import styled from 'styled-components';

import { colors } from 'theme';

export const Heading = styled.h1`
  font-size: 1rem;
  color: ${colors.text};
  text-transform: uppercase;
`;
