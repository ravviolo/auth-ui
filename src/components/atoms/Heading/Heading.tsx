interface Props {
  title: string;
}

export const Heading = ({ title }: Props) => {
  return <h1>{title}</h1>;
};
