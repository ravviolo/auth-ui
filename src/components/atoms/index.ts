export { Button } from './Button/Button';
export { Checkbox } from './Checkbox/Checkbox';
export { Divider } from './Divider/Divider';
export { Heading } from './Heading/Heading';
export { IconButton } from './IconButton/IconButton';
export { InputField } from './InputField/InputField';
export { InputLabel } from './InputLabel/InputLabel';
export { TextButton } from './TextButton/TextButton';
