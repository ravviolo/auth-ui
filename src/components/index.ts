export * from './templates/LoginTemplate/LoginTemplate';
export * from './templates/SignUpTemplate/SignUpTemplate';
