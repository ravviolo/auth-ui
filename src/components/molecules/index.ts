export { AuthFooter } from './AuthFooter/AuthFooter';
export { AuthHeader } from './AuthHeader/AuthHeader';
export { AuthSocials } from './AuthSocials/AuthSocials';
export { EmailInput } from './EmailInput/EmailInput';
export { PasswordInput } from './PasswordInput/PasswordInput';
