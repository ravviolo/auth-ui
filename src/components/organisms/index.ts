export * from './LoginForm/LoginForm';
export * from './SignUpForm/SignUpForm';
