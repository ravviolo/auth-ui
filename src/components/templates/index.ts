export * from './LoginTemplate/LoginTemplate';
export * from './SignUpTemplate/SignUpTemplate';
