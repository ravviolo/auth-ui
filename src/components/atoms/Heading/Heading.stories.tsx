import { ComponentMeta, ComponentStory } from '@storybook/react';

import { Heading } from 'components/atoms';

export default {
  title: 'Atoms/Heading',
  component: Heading,
} as ComponentMeta<typeof Heading>;

const Template: ComponentStory<typeof Heading> = (args) => (
  <Heading {...args} />
);

export const Default = Template.bind({});
Default.args = {
  title: 'I am a heading',
};
