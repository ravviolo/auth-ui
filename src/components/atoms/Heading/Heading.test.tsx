import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';

import { Heading } from 'components/atoms';

import { Props as HeadingProps } from './Heading';

describe('Heading', () => {
  const props: HeadingProps = {
    testId: 'test-heading',
    title: 'Test Heading',
  };

  it('should display text received via props', () => {
    render(<Heading {...props} />);

    const heading = screen.getByTestId('test-heading');

    expect(heading).toHaveTextContent('Test Heading');
  });

  it('should match snapshot', () => {
    const tree = create(<Heading {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
