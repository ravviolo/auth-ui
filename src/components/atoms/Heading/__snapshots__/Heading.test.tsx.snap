// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should match snapshot 1`] = `
<h1
  className="sc-gKXOVf zAjwy"
  data-testid="test-heading"
>
  Test Heading
</h1>
`;
